sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java

```diff
@@ -88,11 +88,19 @@ public void setNextRow(Row row) {
 
     @Override
     public void allFinished(boolean interrupted) {
+        BytesRestResponse response;
         try {
-            channel.sendResponse(new BytesRestResponse(RestStatus.OK, finishBuilder()));
+            response = new BytesRestResponse(RestStatus.OK, finishBuilder());
+        } catch (Throwable t) {
+            fail(t);
+            return;
+        }
+
+        try {
+            channel.sendResponse(response);
```

Comment:
shouldn't it only call `super.allFinished()` once the response is sent like we do in https://github.com/crate/crate/pull/5647/files#diff-e2b04e5afa547e6124e93ca8041c8cd3R57 or does it not matter in the HTTP case?